###**Description :**

Identify the level of income qualification needed for the families in Latin America.

####**Problem Statement Scenario:**
Many social programs have a hard time ensuring that the right people are given enough aid. It's tricky when a program focuses on the poorest segment of the population. This segment of the population can't provide the necessary income and expense records to prove that they qualify. 

In Latin America, a popular method called Proxy Means Test (PMT) uses an algorithm to verify income qualification. With PMT, agencies use a model that considers a family's observable household attributes like the material of their walls and ceiling or the assets found in their homes to classify them and predict their level of need. While this is an improvement, accuracy remains a problem as the region's population grows and poverty declines. 

The Inter-American Development Bank (IDB) believes that new methods beyond traditional econometrics, based on a dataset of Costa Rican household characteristics, might help improve PMT's performance.

**Import Libraries**

In [35]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib in line
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

UsageError: unrecognized arguments: line


**Upload dataset**

In [36]:
data_train = pd.read_csv('train.csv')
data_test = pd.read_csv('test.csv')

In [37]:
data_train.head()

Id      v2a1  hacdor  ...  SQBmeaned  agesq  Target
0  ID_279628684  190000.0       0  ...      100.0   1849       4
1  ID_f29eb3ddd  135000.0       0  ...      144.0   4489       4
2  ID_68de51c94       NaN       0  ...      121.0   8464       4
3  ID_d671db89c  180000.0       0  ...      121.0    289       4
4  ID_d56d6f5f5  180000.0       0  ...      121.0   1369       4

[5 rows x 143 columns]

In [38]:
data_test.head()

Id      v2a1  hacdor  ...  SQBdependency  SQBmeaned  agesq
0  ID_2f6873615       NaN       0  ...           0.25     272.25     16
1  ID_1c78846d2       NaN       0  ...           0.25     272.25   1681
2  ID_e5442cf6a       NaN       0  ...           0.25     272.25   1681
3  ID_a8db26a79       NaN       0  ...           0.00     256.00   3481
4  ID_a62966799  175000.0       0  ...          64.00        NaN    324

[5 rows x 142 columns]

**Exploring dataset**

In [39]:
print('Shape of train data is {}'.format(data_train.shape))
print('Shape of test data is {}'.format(data_test.shape))

Shape of train data is (9557, 143)
Shape of test data is (23856, 142)


**Identifying Target Variable**

In [40]:
for i in data_train.columns:
  if i not in data_test.columns:
    print('Target variable is {}'.format(i))

Target variable is Target


**Understanding the type of data**

In [41]:
print(data_train.dtypes.value_counts())

int64      130
float64      8
object       5
dtype: int64


In [42]:
data_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9557 entries, 0 to 9556
Columns: 143 entries, Id to Target
dtypes: float64(8), int64(130), object(5)
memory usage: 10.4+ MB


Looking at the above types we can see that there are mixed data types

In [43]:
# Exploring each different types of dataset
for i in data_train.columns:
  a=data_train[i].dtype
  if a == 'object':
    print(i)

Id
idhogar
dependency
edjefe
edjefa


**Below is Data description for above object variables**

ID = Unique ID

idhogar = Household level identifier

dependency = Dependency rate, calculated = (number of members of the household younger than 19 or older than 64)/(number of member of household between 19 and 64)

edjefe = years of education of male head of household, based on the interaction of escolari (years of education), head of household and gender, yes=1 and no=0

edjefa =  years of education of female head of household, based on the interaction of escolari (years of education), head of household and gender, yes=1 and no=0

In [44]:
# lets drop the id variable
data_train.drop(['Id','idhogar'],axis=1, inplace=True)

In [45]:
data_train['dependency'].value_counts()

yes           2192
no            1747
0.5           1497
2              730
1.5            713
0.33333334     598
0.66666669     487
8              378
0.25           260
3              236
4              100
0.75            98
0.2             90
0.40000001      84
1.3333334       84
2.5             77
5               24
1.25            18
3.5             18
0.80000001      18
2.25            13
0.71428573      12
0.83333331      11
1.75            11
0.22222222      11
1.2             11
0.2857143        9
0.60000002       8
1.6666666        8
0.16666667       7
6                7
Name: dependency, dtype: int64

In [46]:
# lets convert object variables into numerical data
def map(i):
  if i == 'yes':
    return(float(1))
  elif i == 'no':
    return(float(0))
  else:
    return(float(i))

In [47]:
data_train['dependency'] = data_train['dependency'].apply(map)

In [48]:
for i in data_train.columns:
  a=data_train[i].dtype
  if a == 'object':
    print(i)

edjefe
edjefa


In [49]:
data_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9557 entries, 0 to 9556
Columns: 141 entries, v2a1 to Target
dtypes: float64(9), int64(130), object(2)
memory usage: 10.3+ MB


In [50]:
data_train['edjefe'] = data_train['edjefe'].apply(map)
data_train['edjefa'] = data_train['edjefa'].apply(map)

In [51]:
data_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9557 entries, 0 to 9556
Columns: 141 entries, v2a1 to Target
dtypes: float64(11), int64(130)
memory usage: 10.3 MB


- Now all data is in numerical form

In [52]:
data_train.head()

v2a1  hacdor  rooms  hacapo  ...  SQBdependency  SQBmeaned  agesq  Target
0  190000.0       0      3       0  ...            0.0      100.0   1849       4
1  135000.0       0      4       0  ...           64.0      144.0   4489       4
2       NaN       0      8       0  ...           64.0      121.0   8464       4
3  180000.0       0      5       0  ...            1.0      121.0    289       4
4  180000.0       0      5       0  ...            1.0      121.0   1369       4

[5 rows x 141 columns]

###**Identifying varilable with 0 variance**

In [53]:
df = pd.DataFrame(np.var(data_train,0), columns =['variance'])
df.sort_values(by='variance').head(15)
col=list((df[df['variance']==0]).index)
print('Below are the columns with variance 0-', col)

Below are the columns with variance 0- ['elimbasu5']


elimbasu5 : 1 if rubbish disposal mainly by throwing in river, creek or sea.

Interpretation: It is seen from the above code that all values of elimbasu5 is same so there is no variablity in dataset therefor we will drop this variable

###**Check if there are any biases in your dataset.**


In [60]:
contingency_tab=pd.crosstab(data_train['r4t3'],data_train['hogar_total'])
Observed_Values=contingency_tab.values
import scipy.stats
b=scipy.stats.chi2_contingency(contingency_tab)
Expected_Values = b[3]
no_of_rows=len(contingency_tab.iloc[0:2,0])
no_of_columns=len(contingency_tab.iloc[0,0:2])
df=(no_of_rows-1)*(no_of_columns-1)
print("Degree of Freedom:-",df)
from scipy.stats import chi2
chi_square=sum([(o-e)**2./e for o,e in zip(Observed_Values,Expected_Values)])
chi_square_statistic=chi_square[0]+chi_square[1]
print("chi-square statistic:-",chi_square_statistic)
alpha=0.05
critical_value=chi2.ppf(q=1-alpha,df=df)
print('critical_value:',critical_value)
p_value=1-chi2.cdf(x=chi_square_statistic,df=df)
print('p-value:',p_value)
print('Significance level: ',alpha)
print('Degree of Freedom: ',df)
print('chi-square statistic:',chi_square_statistic)
print('critical_value:',critical_value)
print('p-value:',p_value)
if chi_square_statistic>=critical_value:
    print("Reject H0,There is a relationship between 2 categorical variables")
else:
    print("Retain H0,There is no relationship between 2 categorical variables")
    
if p_value<=alpha:
    print("Reject H0,There is a relationship between 2 categorical variables")
else:
    print("Retain H0,There is no relationship between 2 categorical variables")

Degree of Freedom:- 1
chi-square statistic:- 17022.072400560897
critical_value: 3.841458820694124
p-value: 0.0
Significance level:  0.05
Degree of Freedom:  1
chi-square statistic: 17022.072400560897
critical_value: 3.841458820694124
p-value: 0.0
Reject H0,There is a relationship between 2 categorical variables
Reject H0,There is a relationship between 2 categorical variables


In [62]:
contingency_tab=pd.crosstab(data_train['tipovivi3'],data_train['v2a1'])
Observed_Values=contingency_tab.values
import scipy.stats
b=scipy.stats.chi2_contingency(contingency_tab)
Expected_Values = b[3]
no_of_rows=len(contingency_tab.iloc[0:2,0])
no_of_columns=len(contingency_tab.iloc[0,0:2])
df=(no_of_rows-1)*(no_of_columns-1)
print("Degree of Freedom:-",df)
from scipy.stats import chi2
chi_square=sum([(o-e)**2./e for o,e in zip(Observed_Values,Expected_Values)])
chi_square_statistic=chi_square[0]+chi_square[1]
print("chi-square statistic:-",chi_square_statistic)
alpha=0.05
critical_value=chi2.ppf(q=1-alpha,df=df)
print('critical_value:',critical_value)
p_value=1-chi2.cdf(x=chi_square_statistic,df=df)
print('p-value:',p_value)
print('Significance level: ',alpha)
print('Degree of Freedom: ',df)
print('chi-square statistic:',chi_square_statistic)
print('critical_value:',critical_value)
print('p-value:',p_value)
if chi_square_statistic>=critical_value:
    print("Reject H0,There is a relationship between 2 categorical variables")
else:
    print("Retain H0,There is no relationship between 2 categorical variables")
    
if p_value<=alpha:
    print("Reject H0,There is a relationship between 2 categorical variables")
else:
    print("Retain H0,There is no relationship between 2 categorical variables")


Degree of Freedom:- 1
chi-square statistic:- 54.04781105990782
critical_value: 3.841458820694124
p-value: 1.9562129693895258e-13
Significance level:  0.05
Degree of Freedom:  1
chi-square statistic: 54.04781105990782
critical_value: 3.841458820694124
p-value: 1.9562129693895258e-13
Reject H0,There is a relationship between 2 categorical variables
Reject H0,There is a relationship between 2 categorical variables


In [63]:
contingency_tab=pd.crosstab(data_train['v18q'],data_train['v18q1'])
Observed_Values=contingency_tab.values
import scipy.stats
b=scipy.stats.chi2_contingency(contingency_tab)
Expected_Values = b[3]
no_of_rows=len(contingency_tab.iloc[0:2,0])
no_of_columns=len(contingency_tab.iloc[0,0:2])
df=(no_of_rows-1)*(no_of_columns-1)
print("Degree of Freedom:-",df)
from scipy.stats import chi2
chi_square=sum([(o-e)**2./e for o,e in zip(Observed_Values,Expected_Values)])
chi_square_statistic=chi_square[0]+chi_square[1]
print("chi-square statistic:-",chi_square_statistic)
alpha=0.05
critical_value=chi2.ppf(q=1-alpha,df=df)
print('critical_value:',critical_value)
p_value=1-chi2.cdf(x=chi_square_statistic,df=df)
print('p-value:',p_value)
print('Significance level: ',alpha)
print('Degree of Freedom: ',df)
print('chi-square statistic:',chi_square_statistic)
print('critical_value:',critical_value)
print('p-value:',p_value)
if chi_square_statistic>=critical_value:
    print("Reject H0,There is a relationship between 2 categorical variables")
else:
    print("Retain H0,There is no relationship between 2 categorical variables")
    
if p_value<=alpha:
    print("Reject H0,There is a relationship between 2 categorical variables")
else:
    print("Retain H0,There is no relationship between 2 categorical variables")

Degree of Freedom:- 0
chi-square statistic:- 0.0
critical_value: nan
p-value: nan
Significance level:  0.05
Degree of Freedom:  0
chi-square statistic: 0.0
critical_value: nan
p-value: nan
Retain H0,There is no relationship between 2 categorical variables
Retain H0,There is no relationship between 2 categorical variables


Therefore,variables ('v18q','v18q1') have relationship between them. For good result we can use any one of them.

**Conclusion:** Therefore, there is bias in our dataset.

In [64]:
data_train.drop('r4t3',axis=1,inplace=True)

**Check if there is a house without a family head.**

"parentesco1" =1 if household head

In [66]:
data_train.parentesco1.value_counts()

0    6584
1    2973
Name: parentesco1, dtype: int64

In [67]:
pd.crosstab(data_train['edjefa'],data_train['edjefe'])

edjefe  0.0   1.0   2.0   3.0   4.0   5.0   ...  16.0  17.0  18.0  19.0  20.0  21.0
edjefa                                      ...                                    
0.0      435   123   194   307   137   222  ...   134   202    19    14     7    43
1.0       69     0     0     0     0     0  ...     0     0     0     0     0     0
2.0       84     0     0     0     0     0  ...     0     0     0     0     0     0
3.0      152     0     0     0     0     0  ...     0     0     0     0     0     0
4.0      136     0     0     0     0     0  ...     0     0     0     0     0     0
5.0      176     0     0     0     0     0  ...     0     0     0     0     0     0
6.0      947     0     0     0     0     0  ...     0     0     0     0     0     0
7.0      179     0     0     0     0     0  ...     0     0     0     0     0     0
8.0      217     0     0     0     0     0  ...     0     0     0     0     0     0
9.0      237     0     0     0     0     0  ...     0     0     0     0     0     0
10.0      96     0     0     0     0     0  ...     0     0     0     0     0     0
11.0     399     0     0     0     0     0  ...     0     0     0     0     0     0
12.0      72     0     0     0     0     0  ...     0     0     0     0     0     0
13.0      52     0     0     0     0     0  ...     0     0     0     0     0     0
14.0     120     0     0     0     0     0  ...     0     0     0     0     0     0
15.0     188     0     0     0     0     0  ...     0     0     0     0     0     0
16.0     113     0     0     0     0     0  ...     0     0     0     0     0     0
17.0      76     0     0     0     0     0  ...     0     0     0     0     0     0
18.0       3     0     0     0     0     0  ...     0     0     0     0     0     0
19.0       4     0     0     0     0     0  ...     0     0     0     0     0     0
20.0       2     0     0     0     0     0  ...     0     0     0     0     0     0
21.0       5     0     0     0     0     0  ...     0     0     0     0     0     0

[22 rows x 22 columns]

**Interpretation:** Above cross tab shows 0 male head and 0 female head which implies that there are 435 families with no family head.

###**Count how many null values are existing in columns.**

In [68]:
data_train.isna().sum().value_counts()

0       135
5         2
7928      1
6860      1
7342      1
dtype: int64

Now lets Identify number of null values in Target variable

In [69]:
data_train['Target'].isna().sum()                                                     

0

**Interpretation:** There are no null values in Target variable. Now lets identify and fillna of other variable.

In [70]:
x=[]
for i in data_train.columns:
    a=data_train[i].dtype
    if a == 'float64':
        x.append(i)
print(x)

['v2a1', 'v18q1', 'rez_esc', 'dependency', 'edjefe', 'edjefa', 'meaneduc', 'overcrowding', 'SQBovercrowding', 'SQBdependency', 'SQBmeaned']


In [71]:
data_train[x].isna().sum()

v2a1               6860
v18q1              7342
rez_esc            7928
dependency            0
edjefe                0
edjefa                0
meaneduc              5
overcrowding          0
SQBovercrowding       0
SQBdependency         0
SQBmeaned             5
dtype: int64

In [72]:
data_train['v18q1'].value_counts()

1.0    1586
2.0     444
3.0     129
4.0      37
5.0      13
6.0       6
Name: v18q1, dtype: int64

In [75]:
pd.crosstab(data_train['tipovivi1'],data_train['v2a1'])

v2a1       0.0        12000.0    13000.0    ...  855810.0   1000000.0  2353477.0
tipovivi1                                   ...                                 
0                 29          3          4  ...         11          7          2

[1 rows x 157 columns]

In [76]:
pd.crosstab(data_train['v18q1'],data_train['v18q'])

v18q      1
v18q1      
1.0    1586
2.0     444
3.0     129
4.0      37
5.0      13
6.0       6

**Interpretation and action:** 'v2a1', 'v18q1', 'rez_esc' have more than 50% null values, because for v18q1, there are families with their own house so they won't pay rent in that case it should be 0 and similar is for v18q1 there can be families with 0 tablets.

Istead we can drop a column tipovivi3,v18q

tipovivi3, =1 rented

v18q, owns a tablet

as v2a1 alone can show both **as v18q1 alone can show that if respondent owns a tablet or not

In [77]:
data_train['v2a1'].fillna(0,inplace=True)
data_train['v18q1'].fillna(0,inplace=True)

In [78]:
data_train.drop(['tipovivi3', 'v18q','rez_esc','elimbasu5'],axis=1,inplace=True)

In [80]:
data_train['meaneduc'].fillna(np.mean(data_train['meaneduc']),inplace=True)
data_train['SQBmeaned'].fillna(np.mean(data_train['SQBmeaned']),inplace=True)
print(data_train.isna().sum().value_counts())

0    136
dtype: int64


In [81]:
y=[]
for i in data_train.columns:
    a=data_train[i].dtype
    if a == 'int64':
        y.append(i)
print(y)

['hacdor', 'rooms', 'hacapo', 'v14a', 'refrig', 'r4h1', 'r4h2', 'r4h3', 'r4m1', 'r4m2', 'r4m3', 'r4t1', 'r4t2', 'tamhog', 'tamviv', 'escolari', 'hhsize', 'paredblolad', 'paredzocalo', 'paredpreb', 'pareddes', 'paredmad', 'paredzinc', 'paredfibras', 'paredother', 'pisomoscer', 'pisocemento', 'pisoother', 'pisonatur', 'pisonotiene', 'pisomadera', 'techozinc', 'techoentrepiso', 'techocane', 'techootro', 'cielorazo', 'abastaguadentro', 'abastaguafuera', 'abastaguano', 'public', 'planpri', 'noelec', 'coopele', 'sanitario1', 'sanitario2', 'sanitario3', 'sanitario5', 'sanitario6', 'energcocinar1', 'energcocinar2', 'energcocinar3', 'energcocinar4', 'elimbasu1', 'elimbasu2', 'elimbasu3', 'elimbasu4', 'elimbasu6', 'epared1', 'epared2', 'epared3', 'etecho1', 'etecho2', 'etecho3', 'eviv1', 'eviv2', 'eviv3', 'dis', 'male', 'female', 'estadocivil1', 'estadocivil2', 'estadocivil3', 'estadocivil4', 'estadocivil5', 'estadocivil6', 'estadocivil7', 'parentesco1', 'parentesco2', 'parentesco3', 'parentesco

In [82]:
data_train[y].isna().sum().value_counts()

0    126
dtype: int64

**Interpretation:** Now there is no null value in our datset.

In [83]:
data_train.Target.value_counts()

4    5996
2    1597
3    1209
1     755
Name: Target, dtype: int64

####**Set the poverty level of the members and the head of the house same in a family.**
Now for people below poverty level can be people paying less rent and don't own a house. and it also depends on whether a house is in urban area or rural area.

In [84]:
Poverty_level=data_train[data_train['v2a1'] !=0]

In [85]:
Poverty_level.shape

(2668, 136)

In [86]:
poverty_level=Poverty_level.groupby('area1')['v2a1'].apply(np.median)

In [87]:
poverty_level

area1
0     80000.0
1    140000.0
Name: v2a1, dtype: float64


For rural area level if people paying rent less than 8000 is under poverty level.

For Urban area level if people paying rent less than 140000 is under poverty level.

In [89]:
def povert(x):
    if x<8000:
        return('Below poverty level')
    
    elif x>140000:
        return('Above poverty level')
    elif x<140000:
        return('Below poverty level: Ur-ban ; Above poverty level : Rural ')  

In [90]:
c=Poverty_level['v2a1'].apply(povert)

In [91]:
c.shape

(2668,)

In [92]:
pd.crosstab(c,Poverty_level['area1'])

area1                                                 0     1
v2a1                                                         
Above poverty level                                 139  1103
Below poverty level: Ur-ban ; Above poverty lev...  306  1081

**Interpretation:**

- There are total 1242 people above poverty level independent of area whether rural or Urban
- Remaining 1111 people level depends on their area

**Rural:**

Above poverty level= 445

**Urban:**

Above poverty level =1103

Below poverty level=1081

In [96]:
data_train.head()

v2a1  hacdor  rooms  hacapo  ...  SQBdependency  SQBmeaned  agesq  Target
0  190000.0       0      3       0  ...            0.0      100.0   1849       4
1  135000.0       0      4       0  ...           64.0      144.0   4489       4
2       0.0       0      8       0  ...           64.0      121.0   8464       4
3  180000.0       0      5       0  ...            1.0      121.0    289       4
4  180000.0       0      5       0  ...            1.0      121.0   1369       4

[5 rows x 136 columns]

In [93]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split

In [97]:
X_data=data_train.drop('Target',axis=1)
Y_data=data_train.Target

In [98]:
X_data_col=X_data.columns

####**Applying Standard Scalling to dataset**

In [99]:
from sklearn.preprocessing import StandardScaler
SS=StandardScaler()
X_data_1=SS.fit_transform(X_data)
X_data_1=pd.DataFrame(X_data_1,columns=X_data_col)

####**Now we will proceed to model fitting**

In [100]:
X_train,X_test,Y_train,Y_test=train_test_split(X_data_1,Y_data,test_size=0.25,stratify=Y_data,random_state=0)

Lets identify best parameters for our model using GridSearchCv

In [101]:
from sklearn.pipeline import Pipeline
from sklearn.model_selection import GridSearchCV

rfc=RandomForestClassifier(random_state=0)
parameters={'n_estimators':[10,50,100,300],'max_depth':[3,5,10,15]}
grid=zip([rfc],[parameters])

best_=None

for i, j in grid:
    a=GridSearchCV(i,param_grid=j,cv=3,n_jobs=1)
    a.fit(X_train,Y_train)
    if best_ is None:
        best_=a
    elif a.best_score_>best_.best_score_:
        best_=a
        
        
print ("Best CV Score",best_.best_score_)
print ("Model Parameters",best_.best_params_)
print("Best Estimator",best_.best_estimator_)

Best CV Score 0.8507046183898423
Model Parameters {'max_depth': 15, 'n_estimators': 300}
Best Estimator RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='gini', max_depth=15, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=300,
                       n_jobs=None, oob_score=False, random_state=0, verbose=0,
                       warm_start=False)


In [102]:
RFC=best_.best_estimator_
Model=RFC.fit(X_train,Y_train)
pred=Model.predict(X_test)

In [103]:
print('Model Score of train data : {}'.format(Model.score(X_train,Y_train)))
print('Model Score of test data : {}'.format(Model.score(X_test,Y_test)))

Model Score of train data : 0.9831170643225896
Model Score of test data : 0.8824267782426778


In [104]:
Important_features=pd.DataFrame(Model.feature_importances_,X_data_col,columns=['feature_importance'])

In [105]:
Top50Features=Important_features.sort_values(by='feature_importance',ascending=False).head(50).index

In [106]:
Top50Features

Index(['SQBmeaned', 'meaneduc', 'SQBdependency', 'dependency', 'overcrowding',
       'SQBovercrowding', 'qmobilephone', 'SQBhogar_nin', 'SQBedjefe',
       'edjefe', 'hogar_nin', 'rooms', 'cielorazo', 'r4t1', 'v2a1', 'edjefa',
       'agesq', 'r4m3', 'r4h2', 'SQBage', 'age', 'escolari', 'r4t2', 'r4h3',
       'hogar_adul', 'SQBescolari', 'eviv3', 'bedrooms', 'r4m1', 'epared3',
       'r4m2', 'tamviv', 'paredblolad', 'v18q1', 'SQBhogar_total', 'tamhog',
       'hhsize', 'hogar_total', 'pisomoscer', 'etecho3', 'r4h1', 'lugar1',
       'eviv2', 'tipovivi1', 'energcocinar2', 'energcocinar3', 'epared2',
       'television', 'area2', 'area1'],
      dtype='object')

In [107]:
for i in Top50Features:
    if i not in X_data_col:
        print(i)

In [108]:
X_data_Top50=X_data[Top50Features]

In [109]:
X_train,X_test,Y_train,Y_test=train_test_split(X_data_Top50,Y_data,test_size=0.25,stratify=Y_data,random_state=0)

In [110]:
Model_1=RFC.fit(X_train,Y_train)
pred=Model_1.predict(X_test)

In [111]:
from sklearn.metrics import confusion_matrix,f1_score,accuracy_score  

In [112]:
confusion_matrix(Y_test,pred)

array([[ 143,   17,    0,   29],
       [   8,  324,    4,   63],
       [   1,   12,  214,   75],
       [   2,   10,    3, 1485]])

In [113]:
f1_score(Y_test,pred,average='weighted')

0.9026906492316511

In [114]:
accuracy_score(Y_test,pred)

0.906276150627615

**Interpretation:** Above is our prediction for test data.

**Conclusion:**
Using RandomForest Classifier we can predict test_data with accuracy of 90%.